io/plugins/eu.esdihumboldt.hale.io.jdbc.mssql/src/eu/esdihumboldt/hale/io/jdbc/mssql/MsSqlGeometries.java

```diff
@@ -0,0 +1,231 @@
+					// columnDataType instances? because this type represents
+					// data in a Euclidean (flat) coordinate system.
+
+					// getting Axis order of geometry CRS
+					CRS.AxisOrder axisOrder = CRS.getAxisOrder(geom.getCRSDefinition().getCRS());
+					// if axis order is not x/y ordering then will need to
+					// transform geometry to target crs with longiture first
+					if (axisOrder != CRS.AxisOrder.EAST_NORTH) {
+						if (srsName != null && srsName.startsWith("EPSG")) {
+							targetCRS = CRS.decode(srsName, true);
+						}
+					}
+
+					try {
+						if (srsName != null) {
+							final int index = srsName.lastIndexOf(':');
+							if (index > 0) {
+								srsName = srsName.substring(index + 1).trim();
+							}
+							srId = Integer.parseInt(srsName);
+						}
+					} catch (NumberFormatException nfEx) {
+						// TODO::Using UI ask user to enter default SRId, if
+						// can not extract it.
+					}
+				}
+
+				Geometry targetGeometry;
+				if (targetCRS != null) {
+					MathTransform transform = CRS
+							.findMathTransform(geom.getCRSDefinition().getCRS(), targetCRS);
+					targetGeometry = JTS.transform(geom.getGeometry(), transform);
+				}
+				else {
+					targetGeometry = geom.getGeometry();
+				}
+
+				String sqlForBinaryValue = "DECLARE @g " + columnDataType + ";SET @g = "
+						+ columnDataType + "::STGeomFromText('" + targetGeometry.toText() + "',"
+						+ srId + ");SELECT @g;";
+
+				stmt = connection.createStatement();
+				rs = stmt.executeQuery(sqlForBinaryValue);
+				if (rs.next()) {
+					return rs.getString(1);
+				}
+			} finally {
+				if (rs != null)
+					try {
+						rs.close();
+					} catch (Exception e) {
+						//
+					}
+				if (stmt != null)
+					try {
+						stmt.close();
+					} catch (Exception e) {
+						//
+					}
+			}
+		}
+		return null;
+
+	}
+
+	@Override
+	public GeometryProperty<?> convertToInstanceGeometry(Object geom, TypeDefinition columnType,
+			SQLServerConnection connection) throws Exception {
+
+		Statement stmt = null;
+		ResultSet rs = null;
+		try {
+
+			// We need Column Data type
+			String columnDataType = columnType.getName().getLocalPart();
+
+			String geomAsHex = BaseEncoding.base16().lowerCase().encode((byte[]) geom);
+			String sqlGeom = "SELECT top 1 GeomConvert.geom.STSrid srid, GeomConvert.geom.STAsText() as geomAsText " //
+					+ "FROM " //
+					+ "(SELECT cast(cast(temp.wkb as varbinary(max)) as " + columnDataType
+					+ ") as geom "//
+					+ "FROM " //
+					+ "( select " + "0x" + geomAsHex + " as wkb) as temp" //
+					+ ") " //
+					+ "as GeomConvert"; //
+
+			stmt = connection.createStatement();
+
+			rs = stmt.executeQuery(sqlGeom);
+			Geometry jtsGeom = null;
+			int srId = 0;
+
+			if (rs.next()) {
+				srId = rs.getInt(1);
+				String geomAsText = rs.getString(2);
+				// conversion to JTS via WKT
+				WKTReader wktReader = new WKTReader(new GeometryFactory());
+				try {
+					jtsGeom = wktReader.read(geomAsText);
```

Comment:
Can you state if this covers all geometry types available in MS SQL. Are there any that we don't support?